Sources/Utils.php

```diff
@@ -1255,14 +1255,14 @@ function ($value, $key) use (&$length) {
 	/**
 	 * Wrapper function for json_decode() with error handling.
 	 *
-	 * @param string $json The string to decode.
+	 * @param mixed $json The string to decode.
 	 * @param bool $associative Whether to force JSON objects to be returned as
 	 *    associative arrays. SMF nearly always wants this to be true, but for
 	 *    the sake of consistency with json_decode(), the default is false.
 	 * @param bool $should_log Whether to log errors. Default: true.
 	 * @return mixed The decoded data.
 	 */
-	public static function jsonDecode(string $json, bool $associative = false, bool $should_log = true)
+	public static function jsonDecode(mixed $json, bool $associative = false, bool $should_log = true)
```

Comment:
When would this ever be something besides a string, and why would we want to accept a non-string value here?

Reply:
It was happening on the board index. a int/bool was getting in there.  Best to let it just realize it doesn't need to decode those and return.

Reply:
It would be better to cast the value to string before passing it in. Utils::jsonDecode() is supposed to accept the same input (for the first two parameters) as json_decode().

Reply:
I agree. I didn't add the is_string checks, it was already in there. So we have been already accepting it like this. We would have to adapt the logic further and not send it to this function if it's not a string. If it's an int/bool/float, et don't need to do anything.

Reply:
Also to add:
>  Returns the value encoded in json as an appropriate PHP type. Unquoted values true, false and null are returned as true, false and null respectively. null is returned if the json cannot be decoded or if the encoded data is deeper than the nesting limit. 

We should make jsonDecode and then return null instead of [] on error if we are wanting to match specs here.
I can align that.  Maybe best to do that in a separate PR because its possible it will break things.

What do you think/want?
I have no problem with moving things back into the get to check for non strings and skipping json decode in that case.

Reply:
> We should make jsonDecode and then return null instead of [] on error if we are wanting to match specs here. I can align that. Maybe best to do that in a separate PR because its possible it will break things.
> 
> What do you think/want? I have no problem with moving things back into the get to check for non strings and skipping json decode in that case.

That sounds like an excellent plan. Please do.

Reply:
I aligned it.  I also aligned jsonEncode (the best I could since the return false type isn't supported in 8.0).

To align it, I had to update calls to jsonDecode so I could add the parameters required.  I added a wrapper for the legacy $smcFunc calls.